astroquery/esasky/tests/test_esasky_remote.py

```diff
@@ -78,13 +78,15 @@ def test_esasky_get_images_obs_id(self, tmp_path, mission, obsid):
         result = ESASky.get_images(observation_ids=obsid,
                                    missions=mission, download_dir=tmp_path)
 
-        assert Path(tmp_path, mission).exists()
+        assert Path(tmp_path, mission.upper()).exists()
         if mission == "Herschel":
             assert isinstance(result[mission.upper()][0]["250"], HDUList)
             assert isinstance(result[mission.upper()][0]["350"], HDUList)
             assert isinstance(result[mission.upper()][0]["500"], HDUList)
         else:
             assert isinstance(result[mission.upper()][0], HDUList)
+            for hdu_list in result[mission.upper()]:
+                hdu_list.close()
```

Comment:
Thanks.  I wonder whether this should rather be dealt with by a context manager somewhere in the module instead, but I suppose getting a workaround in the test would do for now.